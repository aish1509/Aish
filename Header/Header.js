import React from 'react';
import PropTypes from 'prop-types';
import {HeaderText, M180Header} from '@math/common';
import {ZoneHeader} from '@math/m180';
import history from '../../utils/history';

const Header = ({pathname}) => {
	const handleNavigation = () => history.push('/welcome');
	switch (pathname) {
		case '/cat':
			return (
				<M180Header firstName="firstname" lastName="lastname">
					Question Validation Tool
				</M180Header>
			);
		case '/explorezone':
			return (
				<ZoneHeader
					firstName="firstname"
					lastName="lastname"
					zoneName="EXPLORE_ZONE"
					onHomeClick={handleNavigation}
				>
					<HeaderText subheaderText="Block 1: Multiplicative Thinking">
						Explore Zone
					</HeaderText>
				</ZoneHeader>
			);
		case '/explorezone/anchorvideo':
			return (
				<ZoneHeader
					firstName="firstname"
					lastName="lastname"
					zoneName="EXPLORE_ZONE"
					onHomeClick={handleNavigation}
				>
					<HeaderText subheaderText="Block 1: Multiplicative Thinking">
						Explore Zone
					</HeaderText>
				</ZoneHeader>
			);

		case '/explorezone/simulation':
			return (
				<ZoneHeader
					firstName="firstname"
					lastName="lastname"
					zoneName="EXPLORE_ZONE"
					onHomeClick={handleNavigation}
				>
					<HeaderText subheaderText="Block 1: Multiplicative Thinking">
						Explore Zone
					</HeaderText>
				</ZoneHeader>
			);

		default:
			return (
				<M180Header firstName="firstname" lastName="lastname">
					MATH 180
				</M180Header>
			);
	}
};

Header.propTypes = {
	pathname: PropTypes.string.isRequired
};

export default Header;
