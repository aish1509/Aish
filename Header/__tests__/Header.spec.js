import {mount, shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import React from 'react';
import history from '../../history';

import Header from '../index';

jest.mock('../../history');

describe('<Header />', () => {
	let wrapper = null;

	describe('mount component', () => {
		beforeEach(() => {
			wrapper = shallow(<Header pathname="/explorezone" />);
		});

		it('watch snapshot', () => {
			expect(shallowToJson(wrapper)).toMatchSnapshot();
		});

		it('verify when pathname is cat', () => {
			wrapper.setProps({pathname: '/cat'});
			expect(shallowToJson(wrapper)).toMatchSnapshot();
		});

		it('verify when pathname is /explorezone/anchorvideo', () => {
			wrapper.setProps({pathname: '/explorezone/anchorvideo'});
			expect(shallowToJson(wrapper)).toMatchSnapshot();
		});

		it('verify when pathname is /explorezone/simulation', () => {
			wrapper.setProps({pathname: '/explorezone/simulation'});
			expect(shallowToJson(wrapper)).toMatchSnapshot();
		});
	});

	describe('mount component', () => {
		beforeEach(() => {
			wrapper = mount(<Header pathname="/explorezone" />);
		});

		it('history.push should have been called with /welcome', () => {
			const IconButtonComp = wrapper.find(
				'WithStyles(ForwardRef(IconButton))'
			);
			expect(IconButtonComp).toHaveLength(1);
			IconButtonComp.simulate('click');
			expect(history.push).toHaveBeenCalledWith('/welcome');
		});
	});
});
