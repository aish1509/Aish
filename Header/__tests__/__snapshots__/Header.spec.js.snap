// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> mount component verify when pathname is /explorezone/anchorvideo 1`] = `
<ZoneHeader
  firstName="firstname"
  lastName="lastname"
  onHomeClick={[Function]}
  onLogout={[Function]}
  onSettingsClick={[Function]}
  showSettings={true}
  zoneName="EXPLORE_ZONE"
>
  <WithStyles(HeaderText)
    subheaderText="Block 1: Rates in Time"
  >
    Explore Zone
  </WithStyles(HeaderText)>
</ZoneHeader>
`;

exports[`<Header /> mount component verify when pathname is /explorezone/simulation 1`] = `
<ZoneHeader
  firstName="firstname"
  lastName="lastname"
  onHomeClick={[Function]}
  onLogout={[Function]}
  onSettingsClick={[Function]}
  showSettings={true}
  zoneName="EXPLORE_ZONE"
>
  <WithStyles(HeaderText)
    subheaderText="Block 1: Multiplicative Thinking"
  >
    Explore Zone
  </WithStyles(HeaderText)>
</ZoneHeader>
`;

exports[`<Header /> mount component verify when pathname is cat 1`] = `
<M180Header
  firstName="firstname"
  lastName="lastname"
  onLogout={[Function]}
  onSettingsClick={[Function]}
  showSettings={false}
>
  Question Validation Tool
</M180Header>
`;

exports[`<Header /> mount component watch snapshot 1`] = `
<ZoneHeader
  firstName="firstname"
  lastName="lastname"
  onHomeClick={[Function]}
  onLogout={[Function]}
  onSettingsClick={[Function]}
  showSettings={true}
  zoneName="EXPLORE_ZONE"
>
  <WithStyles(HeaderText)
    subheaderText="Block 1: Rates in Time"
  >
    Explore Zone
  </WithStyles(HeaderText)>
</ZoneHeader>
`;
